code/datums/enumeration/list_enumerator.dm

```diff
@@ -0,0 +1,36 @@
+/*
+ * Performs unchecked list enumeration.
+ * Not protected against concurrent modification.
+ */
+/datum/enumerator/list
+	///The list we are enumerating over
+	var/list/reference_list
+	///The current index
+	var/current_index = 0
+
+/datum/enumerator/list/New(list/source)
+	. = ..()
+	reference_list = source
+
+///Get the current method
+/datum/enumerator/list/current()
+	if (current_index < 1 || current_index > length(reference_list))
+		return null
+	return reference_list[current_index]
+
+/datum/enumerator/list/has_next()
+	return (current_index + 1) <= length(reference_list)
+
+///Get the next element
+/datum/enumerator/list/next()
+	current_index ++
+	return current()
+
+///Reset back to the start
+/datum/enumerator/list/reset()
+	current_index = 0
```

Comment:
since lists on byond start with one it would make sense if enumerator would do the same

Reply:
Enumerators start at the element before the first index. You have to call next() on them in order to get the first element of the list.